### PR TITLE
[az fleet] Hubless-Hubful PUT and reconcile commands.

### PR DESCRIPTION
**Pull request by Ealianis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
az fleet

File: src/fleet/azext_fleet/custom.py [hunk 1]
```diff
@@ -159,6 +166,59 @@ def update_fleet(cmd,
             managed_service_identity.type = "UserAssigned"
             managed_service_identity.user_assigned_identities = {assign_identity: user_assigned_identity_model()}
 
+    # Currently, patch does not support the HubProfile. Thus fleet update with --enable-hub, will do a PUT.
```
Comment (jim-minter) [inline src/fleet/azext_fleet/custom.py:174]: I don't think we want to do this.

File: src/fleet/azext_fleet/custom.py [hunk 2]
```diff
@@ -211,6 +284,25 @@ def get_credentials(cmd,  # pylint: disable=unused-argument
...
+                    resource_group_name,
+                    name,
+                    no_wait=False):
+
+    poll_interval = 5
+    fleet = client.get(resource_group_name, name)
+    if fleet.hub_profile is not None:
+        poll_interval = 30
+
+    return sdk_no_wait(no_wait,
```
Comment (jim-minter) [inline src/fleet/azext_fleet/custom.py:298]: should set etag here

File: src/fleet/azext_fleet/custom.py [hunk 3]
```diff
@@ -268,6 +360,17 @@ def delete_fleet_member(cmd,  # pylint: disable=unused-argument
...
 
+def reconcile_fleet_member(cmd, # pylint: disable=unused-argument
+                    client,
+                    resource_group_name,
+                    name,
+                    fleet_name,
+                    no_wait=False):
+
+    member = client.get(resource_group_name, fleet_name, name)
+    return sdk_no_wait(no_wait, client.begin_create, resource_group_name, fleet_name, name, member)
```
Comment (jim-minter) [inline src/fleet/azext_fleet/custom.py:371]: should set etag here

File: src/fleet/azext_fleet/custom.py (hunk 1 above)
Comment (Ealianis) [reply src/fleet/azext_fleet/custom.py:174]: Discussing directly.

File: src/fleet/azext_fleet/custom.py (hunk 3 above)
Comment (Ealianis) [reply src/fleet/azext_fleet/custom.py]: Added

File: src/fleet/azext_fleet/tests/latest/test_fleet_hubless_scenario.py [hunk 4]
```diff
@@ -60,13 +60,19 @@ def test_fleet_hubless(self):
             self.check('name', '{fleet_name}')
         ])
 
+        print(1)
```
Comment (jim-minter) [inline src/fleet/azext_fleet/tests/latest/test_fleet_hubless_scenario.py:63]:
```suggestion
```

nit, feel free to ignore and fix next time around

**Review by jim-minter — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**